jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/TriggerAdHocNodeInStageCommand.java

```diff
@@ -57,11 +62,19 @@ public Void execute(Context context) {
         ProcessInstance processInstance = ksession.getProcessInstance(processInstanceId);
         
         DynamicNodeInstance dynamicContext = (DynamicNodeInstance) ((WorkflowProcessInstanceImpl) processInstance).getNodeInstances(true).stream()
-                .filter(ni -> (ni instanceof DynamicNodeInstance) && stageId.equals(ni.getNode().getMetaData().get("UniqueId")))
+                .filter(ni -> (ni instanceof DynamicNodeInstance) && stage.equals(ni.getNode().getMetaData().get("UniqueId")) || stage.equals(ni.getNodeName()))
```

Comment:
Probably missing parentheses? What happens if NodeInstance is not a dynamic one and its nodeName is the one I am looking for? I guess there will be a ClassCastException.
Shouldn't there be
```(ni instanceof DynamicNodeInstance) && (stage.equals(ni.getNode().getMetaData().get("UniqueId")) || stage.equals(ni.getNodeName()))```

Reply:
good catch, fixed